### DISCUSSION TITLE
Integrating a Welcome Page with Nextra in an Existing Next.js App

### DISCUSSION THREAD

**Discussion by eyallampel1** [shuding/nextra#2203] [Q&A]
Hey everyone,

I've recently integrated the Nextra library into my existing Next.js project. I'm trying to set up an initial welcome page in the standard Next.js style. This page will feature a button, and I'd like the Nextra site content to be displayed only after this button is clicked.

Given that I've added Nextra to an already built Next.js app (not started from scratch), does anyone have suggestions or best practices on how to seamlessly incorporate a regular Next.js page before transitioning to the Nextra content? Any insights or advice would be greatly appreciated!

Thanks in advance!

**Comment by caffeine-addictt** [comment]
@eyallampel1

The answer would depend on which NextJS router you are using.
If you are using NextJS App router and are looking to just have nextra handle a top-level path like `/docs/**/*`, you might find this issue helpful https://github.com/shuding/nextra/issues/2688#issuecomment-1972156650.

If you are using the pages router, You can have the root page `/` be a `.tsx/.jsx` file and setting your `_meta.json`. [Nextra layout docs](https://nextra.site/docs/docs-theme/page-configuration#layouts)

```
pages/
    index.tsx
    _meta.json
```

```json
{
  "*": {
    "type": "page"
  },
  "index": {
    "display": "hidden",
    "theme": {
      "layout": "full",
      "timestamp": false
    }
  }
}
```

I've done something similar in the docs of an OSS project of mine. Feel free to reference it [here](https://github.com/python-thread/thread.ngjx.org/blob/staging/src/pages/_meta.json).
